common/src/main/java/org/apache/apex/common/util/JarHelper.java

```diff
@@ -186,4 +190,62 @@ private void addEntry(File file, String name) throws IOException
       jos.closeEntry();
     }
   }
+
+  /**
+   * Returns a full path to the jar-file that contains the given class and all full paths to dependent jar-files
+   * that are defined in the property "apex-dependencies" of the manifest of the root jar-file
+   * @param classPath Class path
+   * @return List of jar-files
+   */
+  public static List<String> getApexDependentJarsByClass(String classPath)
+  {
+    try {
+      return getApexDependentJarsByClass(Thread.currentThread().getContextClassLoader().loadClass(classPath));
+    } catch (ClassNotFoundException ex) {
+      logger.error("Cannot find the class {}", classPath, ex);
+      throw new RuntimeException("Cannot find the class " + classPath, ex);
+    }
+  }
+
+  /**
+   * Returns a full path to the jar-file that contains the given class and all full paths to dependent jar-files
+   * that are defined in the property "apex-dependencies" of the manifest of the root jar-file
+   * @param clazz Class
+   * @return List of jar-files
+   */
+  public static List<String> getApexDependentJarsByClass(Class clazz)
+  {
+    List<String> list = new LinkedList<>();
+
+    try {
+      URLConnection conn = clazz.getResource(clazz.getSimpleName() + ".class").openConnection();
+      if (conn instanceof JarURLConnection) {
+        String jarPath = ((JarURLConnection)conn).getJarFileURL().getFile();
+        list.add(jarPath);
+        logger.debug("Jar-file {} that contains the class {} was added", jarPath, clazz.getCanonicalName());
+        String value = ((JarURLConnection)conn).getMainAttributes().getValue("apex-dependencies");
```

Comment:
The implementation of the method JarURLConnection.getMainAttributes() returns the manifest attributes.

Unfortunately we cannot use the manifest attribute Class-Path, because the purpose of the new property to deploy the corresponded jar-files. But the attribute Class-Path points to the used jars. For instance, you can look at the content of the attribute Class-Path in apex jar-files to see how many jars are included by Maven.